src/Input/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,148 @@
+  );
+  const textFieldInput = getByLabelText(props.label) as HTMLInputElement;
+  return { textFieldInput, utils };
+}
+
+function setupTextFieldWithChangeEvent(otherProps?: any) {
+  const { textFieldInput } = setupTextField(otherProps);
+  userEvent.type(textFieldInput, inputValue);
+  return { textFieldInput };
+}
+
+it('<TextField> should render with a label of Username and type of text', () => {
+  const TextFieldSnapshot = renderer
+    .create(<TextField type={props.type} label={props.label} />)
+    .toJSON();
+  expect(TextFieldSnapshot).toMatchSnapshot();
+});
+
+describe('when it is rendered', () => {
+  it('should display the correct input label', () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput).toBeTruthy();
+  });
+
+  it('should have the correct type', () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput.type).toEqual(props.type);
+  });
+
+  it('should have an empty string as its value', () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput.value).toEqual('');
+  });
+});
+
+describe('when a value is entered', () => {
+  it('should call onChange once', () => {
+    setupTextFieldWithChangeEvent();
+    expect(props.onChange).toHaveBeenCalledTimes(inputValue.length);
+  });
+
+  it('should return the input value when onChange is passed a function: event => event.target.value', () => {
+    setupTextFieldWithChangeEvent();
+    expect(props.onChange.mock.results.slice(-1)[0].value).toEqual(inputValue);
+  });
+
+  it('should display the correct value', () => {
+    const { textFieldInput } = setupTextFieldWithChangeEvent();
+    expect(textFieldInput.value).toEqual(inputValue);
+  });
+});
+
+describe('when status is:', () => {
+  it("'success', it should display a lightblack border", () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput).toHaveStyle(`
+      border-color: ${SecondaryColor.lightblack};
+    `);
+  });
+
+  it("'error', it should display a red border", () => {
+    const { textFieldInput } = setupTextField({ status: 'error' });
+    expect(textFieldInput).toHaveStyle(`
+      border-color: ${PrimaryColor.glintsred};
+    `);
+  });
+});
+
+describe('when disabled is true', () => {
+  it('should display cursor: not-allowed', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
+    expect(textFieldInput).toHaveStyle('cursor: not-allowed;');
+  });
+
+  it('should be disabled', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
+    expect(textFieldInput).toBeDisabled();
+  });
+});
+
+describe('when small is:', () => {
+  it('true, it should display the correct size', () => {
+    const { textFieldInput } = setupTextField({ small: true });
+    expect(textFieldInput).toHaveStyle('padding: 13px 15px;');
+  });
+
+  it('false, it should display the correct size', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
```

Comment:
trivial thing, `disabled: true` is not related to this test case right?

Reply:
Oh yes haha opps my bad